### ISSUE TITLE
konsole does not wait for command to complete when run inside systemd

### ISSUE THREAD

**Issue by boombatower** [steamtricks/steamtricks#30]
I could have sworn that it did, but it definitely does not now. Perhaps after I made it multiple lines with echo, but that would seem unlikely. Anyway it should wait as otherwise it can get ahead of itself.

**Comment by boombatower** [comment]
Some interesting notes.
```bash
xdg-terminal 'sudo zypper ref'
```
waits for the command to complete, but the following does not...in fact it interupts it and it dies after sudo
```bash
xdg-terminal 'sudo zypper ref ; sleep 5'
```

**Comment by boombatower** [comment]
Derp. Which is the whole reason I run commands inside of bash since `xdg-terminal` quits after `stdin` is not read from (or so it seems).
```bash
/usr/bin/env bash -c
```

Unfortunately, that does not explain how this hard to recreate circumstance can occur.

**Comment by boombatower** [comment]
As I already knew...I can reproduce by verifying two games with changes (so an update occurs) that also require a dep check (delete cache for them). For some reason xdg-terminal then seems to not wait like it normally does (or instead perhaps bash is doing something magical to be concurrent)?

So, this is only relevant when a bunch of games update in same time-frame and require dep check...which can occur when a lot of udpates (ie just started steam) or installing a few new apps. It should be fixed, but not exactly sure what is going on.

**Comment by boombatower** [comment]
Seems like this is also the origin of the corrupted log files, the fact that the watch loop seems to be able to jump around.

**Comment by boombatower** [comment]
So to try and isolate this I created the following small script:
```bash
#!/bin/bash

tail -Fn0 "test.log" | \
while read line ; do
  echo "-> $line"
  xdg-terminal '/usr/bin/env bash -c " \
    echo \"sup bro?\"; \
    sudo zypper ref || sleep 5"'
done
```
and ran
```
echo line >> test.log
```

Interestingly, it works exactly as expected...ie it waited for everything to complete and then processed the next line even if I ran a whole bunch before completing the first terminal. Even placing the xdg-terminal in a separate function does not change the result (as expected, but closer to steamtricks).

Something funky is going on. :(

**Comment by boombatower** [comment]
It seems to be impossible to reproduce now...perhaps I was crazy and had a second steamtricks running via systemd. That explanation is also satisfactory since as far as I can tell this should not be able to occur. Will reopen if I can reproduce.

**Comment by boombatower** [comment]
Alright, today I saw this happen with just a single app being verified. One notable piece was that it was running inside of systemd instead of directly.

**Comment by boombatower** [comment]
Alright! That's the difference and explains why I felt this was happening inconsistently only recently. I can run the loop example in place or normal service in systemd and it exhibits the same odd behavior. I am assuming some environment variable indicated interactive mode is not set and thus the terminal runs in the background.

**Comment by boombatower** [comment]
As I suspected after reading through `xdg-terminal` code, this is related to `konsole`. Running directly with konsole shows the same issue and running with `xterm` does not, it waits appropriately.

**Comment by boombatower** [comment]
Looks like I was on the right track as http://stackoverflow.com/questions/1401002/trick-an-application-into-thinking-its-stdin-is-interactive-not-a-pipe works.
```bash
script --return -c "[executable string]" /dev/null
```

I ran the entire script inside that from systemd level, but presumably I could just run the `xdg-terminal` part, just a bit tricky with all the double escaping and not sure which is cleaner.

**Comment by boombatower** [comment]
So interestingly the [systemd `StandardInput`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#StandardInput=) option combined with `TTYPath` do not seem to work and probably won't do well since they require fixed tty path. Presumably they are meant for running X and such to specific tty.

Changing the systemd `ExecStart` to use the `script` approach works exactly as desired where as calling just `xdg-terminal` via `script` inside the loop seems to behave weirdly. Lines added while konsole is open are not handled by script, but are displayed in log output. So for all intensive purposes they are ignored which is arguably worse than current behavior.

For historical reference:
```bash
function faketty { script -qfec "$(printf "%q " "$@")"; }
faketty xdg-terminal ...
```

I guess for now I'll change systemd service file, but that will result in an ugly command... :(

**Comment by boombatower** [comment]
Solved by 351813a.
